sdks/java/core/src/test/java/org/apache/beam/sdk/util/PathUtilsTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.file.Path;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * Unit tests for {@link PathUtils}.
+ */
+public class PathUtilsTest {
```

Comment:
All tests here are for a local-file path. Add tests for GCS- or S3-like paths, with buckets and stuff?

Also, I think the path resolution functions should work on non-existing files too, so you don't need an actual TemporaryFolder, you can hard-code the path of a fictional folder ("/tmp/foo" for example). The tests will probably get much cleaner since you'll drop all those tmpFolder.getRoot().toPath() calls and so on.